app/controllers/response_sets_controller.rb

```diff
@@ -46,6 +47,22 @@ def create
     end
   end
 
+  def transfer_table
```

Comment:
Much of this should be moved into the model itself and not be in the controller.  I would suggest refactoring so the model has an extend method on it.